### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by jaimevillegas**
Hi there!

@SFarida and I have made the following changes to the repository:

## Add Integration tests:
- `post_index_spec.rb`
- `post_show_spec.rb`
- `user_index_spec.rb`
- `user_show_spec.rv`

## Work on N+1 Problem
---
Screenshot before working on the problem:
Captura de pantalla 2023-06-19 151620

Screenshot after working on the problem:
Captura de pantalla 2023-06-19 152057

Thank you so much for taking the time to review our code! 💯 🚀

### CODE REVIEW

File: spec/views/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  it 'displays number of posts' do
+    expect(page).to have_content('Number of Posts: 5')
+  end
+
+  it "displays the user's first 3 posts" do
+    expect(page).to have_content('Post - 1')
+    expect(page).to have_content('Post - 2')
+    expect(page).to have_content('Post - 3')
+  end
```
Comment (zainsadaqat) [inline spec/views/user_show_spec.rb:59]: Dear @jaimevillegas and @SFarida,
Good job for writing test cases However there're a few test cases that are missing: 
1. Tests that assert you can see the user's bio.
2. Tests that asserts you can see a button that lets me view all of a user's posts.
3. Tests that redirect to the post's show page when you click on a user's post.
4. Tests that redirect to the user's post's index page when you click on a **see all posts**.

You can get help from this cheatsheet as well: https://gist.github.com/zhengjia/428105

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
## Hello @jaimevillegas and @SFarida  👋 🙂

You're doing a great job on your project. We're proud of you. Good job so far! 

It's OK to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job. 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Flow
  ✅  Descriptive Pull Request
  ✅  No linter errors

## Required Changes ♻️



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_show_spec.rb (hunk 1 above)
Comment (jaimevillegas) [reply spec/views/user_show_spec.rb:59]: Hello @zainsadaqat 

Thank you for your review! We have worked on solving that!

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    expect(page).to have_content('Post - 2')
+    expect(page).to have_content('Post - 3')
+    expect(page).to have_content('Post - 4')
+  end
+
+  it 'Redirects to the user posts index page when clicking See all posts' do
+    click_on 'See all posts'
+    expect(page).to have_current_path(user_posts_path(@user.id))
+  end
+end
```
Comment (addod19) [inline spec/views/user_show_spec.rb:60]: - [x] Great job adding the tests for `1, 2, and 4`, however, the `3` is missing the tests that redirect to the post's show page when you click on a user's post
Screenshot from 2023-06-19 21-50-55

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @jaimevillegas and @SFarida, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great job adding some of the missing test :rocket: awesome


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_show_spec.rb (hunk 2 above)
Comment (jaimevillegas) [reply spec/views/user_show_spec.rb:60]: Hello @addod19!

Thank you for the comment about the test. We forget to add the line `visit user_posts_path(@user.id))` on the test. It is done now!

**Review by addod19 — APPROVED** [review]
### Hi @jaimevillegas and @SFarida, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [x] Awesome job making all changes :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**